src/openforms/submissions/templates/report/submission_report.html

```diff
@@ -40,24 +40,27 @@ <h1 class="title">{{ report.form.name }}</h1>
                 {% endcomment %}
             </section>
 
-            {% for step in report.steps %}
-                <div class="submission-step">
-                    <h2 class="subtitle">{{ step.title }}</h2>
+            {% for submission_step_node in report.renderer.get_children %}
+                {% if submission_step_node.has_children %}
+                    <div class="submission-step">
+                        <h2 class="subtitle">{{ submission_step_node.render }}</h2>
+                        {% for component_node in submission_step_node.get_children %}
+                            {% if component_node.label or component_node.display_value %}
+                                <div class="submission-step-row {% if component_node.layout_modifier %}submission-step-row--{{ component_node.layout_modifier }}{% endif %}">
 
-                    {% for item in step %}
-                        <div class="submission-step-row {% if item.layout_modifier %}submission-step-row--{{ item.layout_modifier }}{% endif %}">
+                                    <div class="submission-step-row__label">
+                                        {{ component_node.label }}
```

Comment:
At the moment, fieldsets that have been with the option `hideHeader` checked will still show the header

Reply:
oops! good one to catch in the unit test for `FieldSetNode`

Reply:
fixed in 02bdd3c92ead0771e91bdbe097b13ec23ab326bf